beetsplug/spotify.py

```diff
@@ -3,54 +3,319 @@
 from __future__ import division, absolute_import, print_function
 
 import re
+import json
+import base64
 import webbrowser
+
 import requests
-from beets.plugins import BeetsPlugin
-from beets.ui import decargs
+
 from beets import ui
-from requests.exceptions import HTTPError
+from beets.plugins import BeetsPlugin
+from beets.util import confit
+from beets.autotag.hooks import AlbumInfo, TrackInfo, Distance
 
 
 class SpotifyPlugin(BeetsPlugin):
-
-    # URL for the Web API of Spotify
-    # Documentation here: https://developer.spotify.com/web-api/search-item/
-    base_url = "https://api.spotify.com/v1/search"
-    open_url = "http://open.spotify.com/track/"
-    playlist_partial = "spotify:trackset:Playlist:"
+    # Base URLs for the Spotify API
+    # Documentation: https://developer.spotify.com/web-api
+    oauth_token_url = 'https://accounts.spotify.com/api/token'
+    open_track_url = 'http://open.spotify.com/track/'
+    search_url = 'https://api.spotify.com/v1/search'
+    album_url = 'https://api.spotify.com/v1/albums/'
+    track_url = 'https://api.spotify.com/v1/tracks/'
+    playlist_partial = 'spotify:trackset:Playlist:'
 
     def __init__(self):
         super(SpotifyPlugin, self).__init__()
-        self.config.add({
-            'mode': 'list',
-            'tiebreak': 'popularity',
-            'show_failures': False,
-            'artist_field': 'albumartist',
-            'album_field': 'album',
-            'track_field': 'title',
-            'region_filter': None,
-            'regex': []
-        })
+        self.config.add(
+            {
+                'mode': 'list',
+                'tiebreak': 'popularity',
+                'show_failures': False,
+                'artist_field': 'albumartist',
+                'album_field': 'album',
+                'track_field': 'title',
+                'region_filter': None,
+                'regex': [],
+                'client_id': 'N3dliiOOTBEEFqCH5NDDUmF5Eo8bl7AN',
+                'client_secret': '6DRS7k66h4643yQEbepPxOuxeVW0yZpk',
+                'tokenfile': 'spotify_token.json',
+                'source_weight': 0.5,
+                'user_token': '',
+            }
+        )
+        self.config['client_secret'].redact = True
+
+        """Path to the JSON file for storing the OAuth access token."""
```

Comment:
Since this isn't at the top of a function, it's not a docstring—a comment would probably be more appropriate.

Reply:
Yea I wasn't sure if the [Sphinx autoattributes](http://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoattribute) still requires docstrings, but looks like they support comments now.